doc/acl/Everflow-test-plan.md

```diff
@@ -0,0 +1,1134 @@
+            "PRIORITY": "9995"
+        },
+        "EF_EGRESS|RULE_6": {
+            "IP_PROTOCOL": "126",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9994"
+        },
+        "EF_EGRESS|RULE_7": {
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9993",
+            "TCP_FLAGS": "0x12/0x12"
+        },
+        "EF_EGRESS|RULE_8": {
+            "L4_SRC_PORT_RANGE": "4672-4681",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9992"
+        },
+        "EF_EGRESS|RULE_9": {
+            "L4_DST_PORT_RANGE": "4672-4681",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9991"
+        },
+        "EF_EGRESS|RULE_10": {
+            "DSCP": "51",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9990"
+        },
+        "EF_EGRESS|RULE_11": {
+            "IN_PORTS": "Ethernet4,Ethernet8",
+            "MIRROR_EGRESS_ACTION": "session_1",
+            "PRIORITY": "9989"
+        }
+    }
+}
+```
+
+### Run test
+
+For each configuration scenario, we need to run all the sub-tests. Everflow sub-tests consists of a number of test cases. Each of the test case is executed with log analyzer enabled, for example:
+
+1. Run loganalyzer 'init' phase
+2. Run a everflow test case
+3. Run loganalyzer 'analyze' phase
+
+Each test case may involve with with one or more classes defined in the PTF script.
+
+#### PTF Test
+
+The everflow test cases eventually call the ptf scripts to do the actual testing. The PTF scripts inject packets into DUT and validate traffic forwarded by DUT.
+
+PTF test will generate traffic between ports and make sure it mirrored according to the configured Everflow session and ACL rules. Depending on the testbed topology and the existing configuration (e.g. ECMP, LAGS, etc) packets may arrive to different ports. Therefore ports connection information will be generated from the minigraph and supplied to the PTF script.
+
+The `EverflowTest` class in everflow_tb_test.py need to be extended to cover IPv6 testing. Need some new methods for sending and validating IPv6 packets.
+
+## Test cases
+
+Each test case will be additionally validated by the loganalyzer utility.
+
+Each test case will add dynamic Everflow ACL rules at the beginning and remove them at the end.
+
+Each test case will run traffic for persistent and dynamic Everflow ACL rules.
+
+Each test case will analyze Everflow packet header and payload (if mirrored packet is equal to original).
+
+### Test case \#1 - Resolved route
+
+#### Test objective
+
+Verify that session with resolved route has active state.
+
+#### Test steps
+
+- Create route that matches session destination IP with unresolved next hop.
+- Resolve route next hop.
+- Send packets that matches each Everflow ACL rule.
+- Verify that packet mirrored to appropriate port.
+- Analyze mirrored packet header.
+- Verify that mirrored packet payload is equal to sent packet.
```

Comment:
What does it mean 'equal to sent packet'? It depends on ingress/egress stage. Please refer to above comment.

Reply:
Sure, I'll update it to make it more accurate.

Reply:
The check here is for payload of mirrored packet. This part should always be equal to sent packet.